File: cpp/p9.cpp
#include "euler.h"

using namespace std;

int main() {
  for( int a = 1; a <= 998; a++ )
    for( int b = 2; b <= 999; b++ ) {
      int c = 1000 - b - a;
			
      if( a * a + b * b == c * c ) {
	cout << a * b * c << endl;
	return 0;
      }
    }
}

File: clojure/p9.clj
(ns p9
  (:use [euler :only (sqr find-first product timed-test)]))

(defn pythagorean? [[a b c]]
  (= (+ (sqr a) (sqr b)) (sqr c)))

(timed-test
 31875000
 (let [triplets (for [a (range 1 (/ 1000 3))
		      b (range (inc a) (/ 1000 2))]
		  [a b (- 1000 a b)])]
   (product (find-first pythagorean? triplets))))
